tools/topology2/buffer.conf

```diff
@@ -0,0 +1,144 @@
+
+#
+# Common pipeline buffer
+#
+# A generic buffer component. All attributes defined herein are namespaced
+# by alsatplg to "buffer.attribute_name"
+#
+# Usage: this component can be used by declaring in the "widgets" field of
+# a parent object. i.e.
+#
+# widgets [
+# 	"buffer.M.N" {
+#		size	"384"
+#		caps	"101"
+#		underrun_permitted	"true"
+#	}
+# ]
+#
+# Where M is pipeline ID and N is a unique integer in the parent object.
+
+DefineSectionWidget."buffer" {
+
+	############### Boiler Plate Start ####################
+
+	#
+	# Argument used to construct pipeline ID (widget index)
+	#
+
+	@args [ INDEX ]
```

Comment:
Do I understand correctly, that `INDEX` also has to be assigned a value somewhere in these topology files? I cannot seem to find that assignment here. How and where do you plan to assign it?